### PR TITLE
fix: resolve dupe button on one-expense report

### PR DESCRIPTION
**Pull request by daledah**
### Fixed Issues

$ https://github.com/Expensify/App/issues/49793
PROPOSAL: https://github.com/Expensify/App/issues/49793#issuecomment-2377146847

### Tests

Precondition:
1. Control WS, delayed submissions set to Weekly, had admin as approver
2. User submit an expense report, admin approved
3. User submit another report with the same amount and merchant as 2.

Steps: 
1. Open the expense report submitted in step 3
2. Verify that: "Review duplicates" button appears.

- [x] Verify that no errors appear in the JS console

### QA Steps

Precondition:
1. Control WS, delayed submissions set to Weekly, enable Approval and have admin as approver
2. User submit an expense report, admin approved
3. User submit another report with the same amount and merchant as 2.

Steps: 
1. Open the expense report submitted in step 3
2. Verify that: "Review duplicates" button appears.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/054daf76-af25-41f2-a6ec-5865ca99df3c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/6bea2e94-fc75-46a6-90bd-0d8dd4a708a6

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/97ebc2d1-ec04-499c-9917-9f3787bbf8fc

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/e6abee6d-0438-48a5-8838-e0b851164fea

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/13b850ed-27be-4cfd-b488-e9afabf95e36

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/0241a5d4-b092-48cc-95c0-41456a9b7f85

</details>

### CODE REVIEW

**Comment by sobitneupane** [comment]
@daledah The "Review Duplicate" button remains visible even after one of the duplicate expenses is deleted.

**Steps to Reproduce:**

- Delete one of the duplicate expenses.
- Open the remaining expense.
- The "Review Duplicate" option is still visible, even though the duplicate error has been removed.

https://github.com/user-attachments/assets/24a2a3c4-f44d-452b-8e12-34096b45dc46

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>MacOS: Chrome / Safari</summary>

**Existing Review Duplicate button in batched expenses:**
Screenshot 2024-10-04 at 11 25 55

**New Review Duplicate button added by this PR:**
Screenshot 2024-10-04 at 11 26 06
Screenshot 2024-10-04 at 11 30 37

Screenshot 2024-10-04 at 12 12 23
Screenshot 2024-10-04 at 12 12 32

</details>

<details>
<summary>iOS: Native</summary>

**Existing Review Duplicate button in batched expenses:**
Screenshot 2024-10-04 at 12 19 26

**New Review Duplicate button added by this PR:**
Screenshot 2024-10-04 at 12 16 46
Screenshot 2024-10-04 at 12 16 56

</details>

<details>
<summary>iOS: mWeb Safari</summary>

**Existing Review Duplicate button in batched expenses:**
Screenshot 2024-10-04 at 12 19 42

**New Review Duplicate button added by this PR:**
Screenshot 2024-10-04 at 12 06 06
Screenshot 2024-10-04 at 12 06 29

</details>

**Comment by sobitneupane** [comment]
@dannymcclain could you please review the screenshots attached [above](https://github.com/Expensify/App/pull/50133#pullrequestreview-2347257097)?

- After the change, the Approve button will appear alongside the Review duplicates button, while [other buttons](https://github.com/Expensify/App/blob/66aefc37f0d38ea2a2f2246cb037db26f3686cba/src/components/MoneyReportHeader.tsx#L353-L393), such as the Submit button, will each appear on a separate row.

- Green "Review duplicates" button is currently being used for batched expenses. Should we use the same for one-expense report as well?

**Comment by sobitneupane** [comment]
@daledah I can no longer reproduce the issue. I’ll follow up if I manage to replicate it again.

**Comment by dannymcclain** [comment]
The screenshots look a bit off to me. I think the `Resolve duplicates` button should be green if it's the only action available. If there is another, higher-priority action (like approve or submit) then it would go to our default/standard button color.

On mobile, the buttons should be next to each other instead of stacked.

image

cc'ing @dubielzyk-expensify & @JmillsExpensify for a gut check on all this.

**Comment by dubielzyk-expensify** [comment]
That looks right to me Danny!

**Comment by pecanoro** [comment]
@daledah Are all the buttons consistent with what @dannymcclain posted?

**Comment by dannymcclain** [comment]
Updated screenshots/videos are looking good.

On mobile, it looks like there's a second or two when the report is loading that the big green button (Approve for example) is full-width, and then the Resolve duplicates button flashes in and the layout changes—is there any way to get rid of that jump or is it just a consequence of how we're loading the state of the report?

Otherwise layout is looking good to me.

**Comment by pecanoro** [comment]
@sobitneupane Ready for you to do another review!

**Comment by sobitneupane** [comment]
- https://github.com/Expensify/App/pull/50133#issuecomment-2402334456
- Padding off for `Review Duplicates` button when it is the only button (gap between button and the search icon)
Screenshot 2024-10-10 at 17 42 01
Screenshot 2024-10-10 at 17 42 10

- When `shouldShowIntegrationButton` is true, `Review Duplicates` button takes more space.
Screenshot 2024-10-10 at 17 38 20

File: src/components/MoneyReportHeader.tsx [hunk 1]
```diff
@@ -247,6 +248,11 @@ function MoneyReportHeader({policy, report: moneyRequestReport, transactionThrea
         [chatReport?.isOwnPolicyExpenseChat, policy?.harvesting?.enabled],
     );
 
+    const shouldDuplicateButtonBeSuccess = useMemo(
+        () => isDuplicate && !shouldShowSettlementButton && !shouldShowExportIntegrationButton && !shouldShowSubmitButton && !hasAllPendingRTERViolations,
```
Comment (sobitneupane) [inline src/components/MoneyReportHeader.tsx:252]: Things to be considered `shouldDuplicateButtonBeSuccess`:
- Looks like the Submit button is not success in employer's end
Screenshot 2024-10-10 at 17 34 32
- I think ExportIntegrationButton will always have default color.

Comment (sobitneupane) [reply src/components/MoneyReportHeader.tsx:252]: I believe so. In that case, should we have Review duplicates button `success`? cc: @dannymcclain

Comment (dannymcclain) [reply src/components/MoneyReportHeader.tsx:252]: 🤔 Hmm, maybe? In this case I'm not sure. I don't think it's wrong to have two regular buttons here. But maybe @JmillsExpensify will have a better informed opinion on if the review duplicates button should be `success` colored in this situation.

Comment (JmillsExpensify) [reply src/components/MoneyReportHeader.tsx:252]: It's not wrong to have two regular buttons. This is the case of delayed submission being enabled, with a submission frequency set.

**Comment by sobitneupane** [comment]
Sorry for the delay. I will try to get it done by EOD.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/user-attachments/assets/f48a5d25-b3c8-40fb-9d6c-ea7f0ad2ea73


</details>

<details>
<summary>iOS: Native</summary>


https://github.com/user-attachments/assets/d38307f6-658a-49d3-b959-0f62c1b1759f


</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/user-attachments/assets/a763b84a-390d-4037-8da3-edbb1f5bea85


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/user-attachments/assets/6c43725d-0c34-4821-9c40-4702387971db


</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/user-attachments/assets/726b3402-d5b2-48a0-9cd8-f001fa1b206c



</details>

**Comment by sobitneupane** [comment]
Thanks for the update @daledah. The changes look good.

However, it seems that the gap between the header and the `MoneyReportHeaderStatusBar`(for nextStep) has increased due to the changes introduced in this PR.
Screenshot 2024-10-16 at 17 49 05
Screenshot 2024-10-16 at 17 49 09

**Comment by pecanoro** [comment]
@sobitneupane If you can check the latest changes and complete the checklist today, it would be great!

**Comment by sobitneupane** [comment]
@daledah There are still some issues with the spacing on small screens. Let's make sure we avoid any unintended UI changes.
Screenshot 2024-10-16 at 21 31 09  Screenshot 2024-10-16 at 21 31 05

**Comment by sobitneupane** [comment]
@daledah There is merge conflict. Can you please resolve it.

**Comment by sobitneupane** [comment]
@daledah [This issue](https://github.com/Expensify/App/pull/50133/#issuecomment-2417234260) is still reproducible.

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Review by pecanoro — APPROVED** [review]

**Comment by Shahidullah-Muffakir** [comment]
Related PR https://github.com/Expensify/App/pull/50133

**Pull request merged**